packages/@aws-cdk/aws-iam/lib/role.ts

```diff
@@ -130,22 +130,63 @@ export interface RoleProps {
  * the specified AWS service principal defined in `serviceAssumeRole`.
  */
 export class Role extends Resource implements IRole {
-
   /**
-   * Imports an external role by ARN
+   * Imports an external role by ARN.
+   * The imported role will allow adding policies and statements to it
+   * (by creating new policy resources in the given scope)
+   * if the stack `scope` belongs to either doesn't have an account set,
+   * or has the same account set as the one used in the `roleArn` parameter.
+   * If the account for the stack is set,
+   * but to a different value than the account in the `roleArn` parameter,
+   * no policy resources will be created,
+   * and the returned role object will ignore all calls to `addToPolicy`
+   * and `attachInlinePolicy`.
+   *
    * @param scope construct scope
    * @param id construct id
    * @param roleArn the ARN of the role to import
    */
   public static fromRoleArn(scope: Construct, id: string, roleArn: string): IRole {
+    const scopeStack = Stack.of(scope);
```

Comment:
But honestly, I don't think I have a better way :(

Reply:
I don't understand this comment...

`fromRoleArn` creates a new Construct with `scope` as the parent. Then, `addToPolicy` creates a new Policy with `this` as the parent, which means it will always be used in the context of this stack. What am I missing here?

Reply:
I suppose you are correct. I was thinking about a convoluted case where the role was imported in a stack, but then the object is shared and used in a different stack. But yes, the created `Policy` object would still live in the imported stack.